src/main/scala/com/databricks/labs/overwatch/pipeline/Snapshot.scala

```diff
@@ -0,0 +1,365 @@
+        }
+
+        val streamManager = Helpers.getQueryListener(streamWriter,workspace.getConfig, workspace.getConfig.auditLogConfig.azureAuditLogEventhubConfig.get.minEventsPerTrigger)
+        spark.streams.addListener(streamManager)
+        val listenerAddedMsg = s"Event Listener Added.\nStream: ${streamWriter.name}\nID: ${streamWriter.id}"
+        logger.log(Level.INFO, listenerAddedMsg)
+
+        streamWriter.awaitTermination()
+        spark.streams.removeListener(streamManager)
+        logger.log (Level.INFO, s"Streaming COMPLETE: ${cloneSpec.source} --> ${cloneSpec.target}")
+        spark.conf.unset("spark.databricks.delta.commitInfo.userMetadata")
+        CloneReport(cloneSpec, s"Streaming For: ${cloneSpec.source} --> ${cloneSpec.target}", "SUCCESS")
+
+      } catch {
+        case e: Throwable if (e.getMessage.contains("is after the latest commit timestamp of")) => {
+          val failMsg = PipelineFunctions.appendStackStrace(e)
+          val msg = s"SUCCESS WITH WARNINGS: The timestamp provided, ${cloneSpec.asOfTS.get} " +
+            s"resulted in a temporally unsafe exception. Cloned the source without the as of timestamp arg. " +
+            s"\nDELTA ERROR MESSAGE: ${failMsg}"
+          logger.log(Level.WARN, msg)
+          CloneReport(cloneSpec, s"Streaming For: ${cloneSpec.source} --> ${cloneSpec.target}", msg)
+        }
+        case e: Throwable => {
+          val failMsg = PipelineFunctions.appendStackStrace(e)
+          CloneReport(cloneSpec, s"Streaming For: ${cloneSpec.source} --> ${cloneSpec.target}", failMsg)
+        }
+      }
+    }).toArray.toSeq
+
+    val cloneReportPath = s"${snapshotRootPath}/clone_report/"
+    cloneReport.toDS.write.mode("append").option("mergeSchema", "true").format("delta").save(cloneReportPath)
+    logger.log(Level.INFO, s"Clone report has been generated to this path: ${cloneReportPath}")
+  }
+
+  private[overwatch] def buildCloneSpecs(
+                                          cloneLevel: String,
+                                          sourcesToSnap: Array[PipelineTable]
+                                        ): Seq[CloneDetail] = {
+
+    val finalSnapshotRootPath  = if (Array("migration","restore").contains(processType.toLowerCase())){
+      s"${snapshotRootPath}/global_share"
+    }else{
+      s"${snapshotRootPath}/data"
+    }
+
+    sourcesToSnap.map(dataset => {
+      val sourceName = dataset.name.toLowerCase
+      val sourcePath = dataset.tableLocation
+      val mode = dataset._mode
+      val immutableColumns = (dataset.keys ++ dataset.incrementalColumns).distinct
+      val targetPath = s"$finalSnapshotRootPath/$sourceName"
+      CloneDetail(sourcePath, targetPath, None, cloneLevel,immutableColumns,mode)
+    }).toSeq
+  }
+
+  private[overwatch] def tableToExclude (sourceToSnap : Array[PipelineTable], excludes: Option[String] = Some("")) : Array[PipelineTable] = {
```

Comment:
the name of this function is misleading. It should reflect what the function actually does

Reply:
Changed it to filteredTables

Reply:
I don't see the name change